extensions/ql-vscode/src/variant-analysis/variant-analysis-manager.ts

```diff
@@ -123,6 +127,18 @@ export class VariantAnalysisManager
     );
   }
 
+  getCommands(): VariantAnalysisCommands {
+    return {
+      "codeQL.openVariantAnalysisLogs": async (variantAnalysisId: number) => {
+        await this.openVariantAnalysisLogs(variantAnalysisId);
+      },
+    };
+  }
+
+  get commandManager(): ExtensionCommandManager {
```

Comment:
Can you explain why this is needed?

Reply:
This is needed to avoid any direct dependencies from the `VariantAnalysisView` to the `VariantAnalysisManager`; it's the implementation of the `VariantAnalysisViewManager` interface.

Reply:
Can't the view get the command manager (or the App container) passed in? Sorry if I'm missing something obvious! Also, this is not blocking.

Reply:
I think that the variant analysis view only has access to the variant analysis manager and not the app (because the app is private), so it needs this to call the commands.

Reply:
I think we could also pass it in, but as @starcke points out we still need to have some way to get the command manager out of the variant analysis manager (in the [variant analysis view serializer](https://github.com/github/vscode-codeql/blob/39d91788dc00a8ed1f8b6daf011e3991680f2e05/extensions/ql-vscode/src/variant-analysis/variant-analysis-view-serializer.ts)), so we still wouldn't be able to get rid of this method. I'd prefer to keep the variant analysis view serializer as lightweight as possible so that we don't need to change the constructor parameters that often.

Reply:
Hmm okay. It feels a bit off to me but we can circle back some other time if it actually becomes a problem.